src/pedal/transparent.rs

```diff
@@ -0,0 +1,26 @@
+use crate::Pedal;
+
+#[derive(Debug)]
+pub struct Transparent {}
+
+impl Transparent {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Pedal for Transparent {
+    fn name(&self) -> String {
+        "Transparent".into()
```

Comment:
There are a few things going on here.

The first is that `"Transparent"`, is not a `String`, it's a string slice (`&str`). (Similar distinction as `Vec` <-> `&[...]`.) Literals like this are slices rather than full strings because they are a reference to a string hard-coded into the binary. Since it's a string slice, we need to convert it into a `String` to return it here.

There is a `From<T>` trait that you can implement when one type can be created from another. `String` implements `From<&str>`, which means it'll expose a `from` static function that produces a string:

```rust
String::from("some string slice")
```

There's another trait, `Into<T>`, which is the inverse of `From`: it converts an existing value into another type. You're not supposed to manually write the implementation of this trait, because it is automatically written based on existing `From` implementations. eg., because `String` implements `From<&str>`, an implementation of `Into<String>` will be created for `&str`. The `Into` trait exposes the `into()` function I used here. The compiler knows the return type of the function, so it calls the correct version of `into()` for `&str`.

Alternatives I could have used:

```rust
String::from("Transparent");
"Transparent".to_string();
```

Those are both a bit more verbose, so I tend to opt for the shorter `into()`, but it can be useful to be more explicit depending on the situation.

Reply:
The docs explain this well:

https://doc.rust-lang.org/std/convert/trait.From.html
https://doc.rust-lang.org/std/convert/trait.Into.html